federation/pkg/dnsprovider/providers/azure/azuredns/stubs/azurednsapi.go

```diff
@@ -0,0 +1,189 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package azuredns a stub for the Azure DNS API. It used  for unit testing purposes
+package azuredns
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/arm/dns"
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/to"
+)
+
+// API abstracts access to the interfaces in the Azure SDK for Go behind mockable methods
+type API interface {
+	ListZones() (dns.ZoneListResult, error)
+	CreateOrUpdateZone(zoneName string, zone dns.Zone, ifMatch string, ifNoneMatch string) (dns.Zone, error)
+	DeleteZone(zoneName string, ifMatch string, cancel <-chan struct{}) (<-chan dns.ZoneDeleteResult, <-chan error)
+	ListResourceRecordSetsByZone(zoneName string) (*[]dns.RecordSet, error)
+	CreateOrUpdateRecordSet(zoneName string, relativeRecordSetName string, recordType dns.RecordType, parameters dns.RecordSet, ifMatch string, ifNoneMatch string) (dns.RecordSet, error)
+	DeleteRecordSet(zoneName string, relativeRecordSetName string, recordType dns.RecordType, ifMatch string) (result autorest.Response, err error)
+}
+
+// Compile time check for interface conformance
+var _ API = &MockAPI{}
+
+// MockAPI is a minimal implementation used for unit testing.
+type MockAPI struct {
+	zones      map[string]*dns.Zone
+	recordSets map[string][]dns.RecordSet
+}
+
+// NewAPIStub returns an initialized AzureDNSAPIStub
+func NewAPIStub() API {
+	api := &MockAPI{
+		zones:      make(map[string]*dns.Zone),
+		recordSets: make(map[string][]dns.RecordSet),
+	}
+	api.zones["test.com"] = &dns.Zone{ID: to.StringPtr("zoneID"), Name: to.StringPtr("test.com"), Type: to.StringPtr("ZoneTypes")}
+	return api
+}
+
+// DeleteRecordSet simulates deleting a record set
+func (a *MockAPI) DeleteRecordSet(zoneName string, relativeRecordSetName string, recordType dns.RecordType, ifMatch string) (autorest.Response, error) {
+	result := autorest.Response{}
+
+	key := zoneName
+
+	_, ok := a.recordSets[key]
+	if ok {
+		delete(a.recordSets, key)
+	} else {
+		// Deleting non-existant item. Some of the tests do that
```

Comment:
In that case better to call this method "EnsureRecordsetDeleted".
Given that this method never returns anything other than an explicit nil for error, why have it as a return value?

Reply:
This was only a mock. It's not supposed to do much.